db, docs: overhaul docstrings & add SQLAlchemy to intersphinx mappings

dgw opened the pull request:
Some of the stuff in `SopelDB` needed to reference SQLAlchemy docs, and adding those intersphinx mappings was trivially easy, so it was done.

Exirel commented on sopel/db.py:
```diff
@@ -42,15 +42,15 @@ def _deserialize(value):
 class NickIDs(BASE):
     """
-    NickIDs SQLAlchemy Class
+    Nick IDs table SQLAlchemy class.
```
This should be on the same line as the opening quotes `"""Nick IDs table SQLAlchemy class."""`.

Exirel commented on sopel/db.py:
```diff
@@ -42,15 +42,15 @@ def _deserialize(value):
 class Nicknames(BASE):
     """
-    Nicknames SQLAlchemy Class
+    Nicknames table SQLAlchemy class.
```
Ditto.

Exirel commented on sopel/db.py:
```diff
@@ -61,7 +61,7 @@ class Nicknames(BASE):
 class NickValues(BASE):
     """
-    NickValues SQLAlchemy Class
+    Nick values table SQLAlchemy class.
```
Ditto.

Exirel commented on sopel/db.py:
```diff
@@ -72,7 +72,7 @@ class NickValues(BASE):
 class ChannelValues(BASE):
     """
-    ChannelValues SQLAlchemy Class
+    Channel values table SQLAlchemy class.
```
Ditto.

Exirel commented on sopel/db.py:
```diff
@@ -83,7 +83,7 @@ class ChannelValues(BASE):
 class PluginValues(BASE):
     """
-    PluginValues SQLAlchemy Class
+    Plugin values table SQLAlchemy class.
```
You get the idea.

Exirel commented on sopel/db.py:
```diff
@@ -93,14 +93,19 @@ class PluginValues(BASE):
+    When configured to use SQLite with a relative filename, the file is assumed
+    to be in the directory named by the core setting ``homedir``.
+
+    .. versionadded:: 5.0
```
I'd add a `.. versionchanged:: 7.0` to talk about the switch to SQLAlchemy here.

dgw replied:
Done, see https://github.com/sopel-irc/sopel/compare/deea18348561879d1fd03e6ec00ac67f1e924327..75d276652e2288f28b7fe6f9805b43c6a7be2dab

dgw commented:
@Exirel It's pretty easy to see exactly what I changed in response to your review [here](https://github.com/sopel-irc/sopel/compare/deea18348561879d1fd03e6ec00ac67f1e924327..75d276652e2288f28b7fe6f9805b43c6a7be2dab).

(Then I rebased in a separate force-push, because #1774 happened and Git couldn't figure out how to handle that one-line change itself. 😁)
